### PR TITLE
Cleanup ViewRegistry, move processor creation to InstrumentProcessor

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/InstrumentProcessor.java [hunk 1]
```diff
@@ -37,41 +37,40 @@
...
+            /* delta= */ false);
+      case DELTA:
+        return new InstrumentProcessor<>(
+            descriptor,
+            aggregator,
+            meterProviderSharedState.getResource(),
+            meterSharedState.getInstrumentationLibraryInfo(),
+            meterProviderSharedState.getClock(),
+            /* delta= */ true);
+    }
```
Comment (breedx-splk) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/InstrumentProcessor.java:72]: The only difference between the two cases is the delta boolean.  If you make a local var like 
```
boolean delta = configuration.getTemporality() == DELTA;
```
you can simplify the switch, remove the `/* delta */` comment, and it cleans up a little bit.

File: sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/ViewRegistryTest.java [hunk 2]
```diff
@@ -7,95 +7,16 @@
...
-
-    InstrumentProcessor<?> result =
-        viewRegistry.createBatcher(providerSharedState, meterSharedState, descriptor);
-
-    assertThat(result.generatesDeltas()).isTrue();
-    assertThat(result).isEqualTo(expectedInstrumentProcessor);
-
-    assertThat(result).isNotNull();
-  }
-
```
Comment (breedx-splk) [inline sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/ViewRegistryTest.java:98]: Did we lose coverage here?  Should this have moved to the InstrumentProcessorTest instead?

**Review by breedx-splk — CHANGES_REQUESTED** [review]
This looks good to me.  The whole `sharedState.getViewRegistry().createBatcher(...)` felt like a law of Demeter violation, so this change simplifies that for sure.  👍 Had a small suggestion and a question around test coverage.

File: sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/ViewRegistryTest.java (hunk 2 above)
Comment (bogdandrutu) [reply sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/ViewRegistryTest.java:98]: Already tested when we test all the instrument types

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/InstrumentProcessor.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/InstrumentProcessor.java]: Good suggestion, done.

**Review by jkwatson — APPROVED** [review]
👍

**Pull request merged**